clients/rust/src/client.rs

```diff
@@ -227,18 +227,21 @@ impl ScopeInner {
         &self.usecase
     }
 
-    fn as_path_segment(&self) -> &str {
-        if self.scope.is_empty() {
-            "_"
-        } else {
-            &self.scope
+    fn as_path_segment(&self) -> String {
+        if self.scopes.is_empty() {
+            return "_".to_string();
         }
+        self.scopes
+            .iter()
+            .map(|scope| format!("{}={}", scope.name(), scope.value()))
+            .collect::<Vec<String>>()
+            .join(";")
```

Comment:
We should use `self.scopes.as_path_segment().to_string()` here since that manages to format without intermediate allocations.